openedx/core/djangoapps/content/search/handlers.py

```diff
@@ -96,6 +97,7 @@ def xblock_deleted_handler(**kwargs) -> None:
 
 
 @receiver(LIBRARY_BLOCK_CREATED)
+@receiver(LIBRARY_BLOCK_UPDATED)
 @only_if_meilisearch_enabled
 def library_block_updated_handler(**kwargs) -> None:
     """
```

Comment:
When I create a new component in the library, it gets indexed correctly and it shows up in the mfe when I refresh. However when I update the block inside the library, the index doesn't get updated. I've looked into this a bit after our discussion, it seems like the `LIBRARY_BLOCK_UPDATED` is actually not being fired.

The only place where I found it in the codebase was here: 
https://github.com/openedx/edx-platform/blob/96da9902c60709ea5a46eb30d3782e4cad9d84b7/openedx/core/djangoapps/content_libraries/api.py#L724-L729

However the encompassing function `set_library_block_olx` isn't actually being called when I update a component inside my library. I think we're probably missing firing the `LIBRARY_BLOCK_UPDATED` in the right spot when updating the library component. I haven't looked into exactly where that would be, but thought I'd mention it in case it helps in narrowing down the issue.

Reply:
@yusuf-musleh Oddly, I don't think we have an API method for updating library block metadata yet. There definitely [isn't support for editing library blocks the frontend](https://github.com/openedx/frontend-app-course-authoring/issues/1040#issuecomment-2135814907) yet either, so I guess that adding this will be part of that scope?

How did you update your library component?

Reply:
@pomegranited Oh I see, I've been using the library authoring mfe to add/edit components inside the libraries, not the course authoring mfe.

The endpoint that the library authoring mfe is calling when updating the library components is [here](https://github.com/openedx/edx-platform/blob/57fd67158cd19294a4366ce09b7573657c8ae97d/openedx/core/djangoapps/xblock/rest_api/views.py#L211). I'm not sure if it is out of scope for this task to add the `LIBRARY_BLOCK_UPDATED` trigger to this endpoint or if that is even something we want to do in the first place, since we are migrating everything to the course authoring mfe and the endpoints called might likely change.

Reply:
cf [b7f8f8e](https://github.com/openedx/edx-platform/pull/35103/commits/b7f8f8e69916a11420de7bdfaea6f88258be2710)